runtime/parsing_functions.cpp

```diff
@@ -0,0 +1,325 @@
+#include "parsing_functions.h"
+
+/*
+ * Helper functions
+ */
+
+string clearQuotes(const string &str)
+{
+  if (str.starts_with(string("\'"))) {
+    return clearSingleQuotes(str);
+  }
+
+  if (str.starts_with(string("\""))) {
+    return clearDoubleQuotes(str);
+  }
+
+  return str;
+}
+
+string clearSingleQuotes(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  if (str.starts_with(string("\'")) && str.ends_with(string("\'"))) {
+    return str.substr(1, str.size()-2);
+  }
+
+  return str;
+}
+
+string clearDoubleQuotes(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  if (str.starts_with(string("\"")) && str.ends_with(string("\""))) {
+    return str.substr(1, str.size()-2);
+  }
+
+  return str;
+}
+
+string clearExtraSpaces(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  string clear_str;
+
+  for (string::size_type i = 0; i < str.size(); i++) {
```

Comment:
When you use postfix increment/decrement (`i++` / `i--`) you make an additional copy. Therefore, it's recommended to use prefix increment/decrement (`++i` / `--i`).

P.S. This comment applies to the entire code.